packages/web-components/src/slider/slider.ts

```diff
@@ -153,75 +367,100 @@ export class Slider extends FormAssociatedSlider implements SliderConfiguration
    * @public
    */
   public get valueAsNumber(): number {
-    return parseFloat(super.value);
+    return parseFloat(this.value);
   }
 
   public set valueAsNumber(next: number) {
     this.value = next.toString();
   }
 
   /**
-   * Custom function that generates a string for the component's "aria-valuetext" attribute based on the current value.
+   * Custom function that generates a string for the component's "ariaValueText" on element internals based on the current value.
    *
    * @public
    */
   @observable
-  public valueTextFormatter: (value: string) => string | null = () => null;
+  public valueTextFormatter: (value: string) => string = () => '';
+  protected valueTextFormatterChanged() {
+    if (typeof this.valueTextFormatter === 'function') {
+      this.elementInternals.ariaValueText = this.valueTextFormatter(this._value);
+    } else {
+      this.elementInternals.ariaValueText = '';
+    }
+  }
 
   /**
-   * @internal
+   * The element's disabled state.
+   * @public
+   * @remarks
+   * HTML Attribute: `disabled`
    */
-  public valueChanged(previous: string, next: string): void {
-    if (this.$fastController.isConnected) {
-      const nextAsNumber = parseFloat(next);
-      const value = limit(this.min, this.max, this.convertToConstrainedValue(nextAsNumber)).toString();
-
-      if (value !== next) {
-        this.value = value;
-        return;
-      }
-
-      super.valueChanged(previous, next);
-
-      this.setThumbPositionForOrientation(this.direction);
-
-      this.$emit('change');
-    }
+  @attr({ mode: 'boolean' })
+  public disabled!: boolean;
+  protected disabledChanged(): void {
+    this.setDisabledSideEffect(this.disabled);
   }
 
   /**
    * The minimum allowed value.
    *
-   * @defaultValue - 0
    * @public
    * @remarks
    * HTML Attribute: min
    */
-  @attr({ converter: nullableNumberConverter })
-  public min: number = 0; // Map to proxy element.
-  protected minChanged(): void {
-    if (this.proxy instanceof HTMLInputElement) {
-      this.proxy.min = `${this.min}`;
+  @attr({ converter: numberLikeStringConverter })
+  public min?: string;
+  protected minChanged(prev: string | undefined, next: string | undefined): void {
+    this.elementInternals.ariaValueMin = `${this.minAsNumber}`;
+    if (this.$fastController.isConnected && this.minAsNumber > this.valueAsNumber) {
+      this.value = this.min!;
     }
+  }
 
-    this.validate();
+  /**
+   * Returns the min property or the default value
+   *
+   * @internal
+   */
+  public get minAsNumber(): number {
```

Comment:
These are used in the tests.

Reply:
> These are used in the tests.

Let me take a look at these tests - if this is not intended to be used as part of the public API, we likely shouldn't be testing it directly. Exceptions are things that need to be bound in the template.

Reply:
So, I think the question here is - should those be part of the public API? If not, we should treat them as internals and not test them directly just for testing. Our testing strategy should only test the public API (perhaps we need to parse the string to a number in the test) and we should treat everything else as internals. This way, if we want to change the method of how we accommodate this, the tests don't break, because the actual public API is not breaking.

Reply:
@marchbox unresolved this to ensure you see my comment :)

Reply:
Yeah, that's fair. I changed them to private.